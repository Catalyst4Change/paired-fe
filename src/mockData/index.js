export const mockUser = {
  name: "Hillary",
  program: "FE",
  module: 4,
  id: "5caa7eebfdebb8348e53a48e"
}

export const mockSchedule = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
];

export const mockAvailPairings = [
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "morning",
    id: "5caa7eebfdebb8348e53a4a4",
  },
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a4a5",
  }
]; 

export const mockPairingToDeleteFromSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Mon Apr 15 2019',
  time: 'lunch',
  id: '5caa7eebfdebb8348e53a490',
  notes: null
};

export const mockScheduleAfterDelete = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
]; 

export const mockPairingToAddToSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Thu Apr 18 2019',
  time: 'morning',
  id: '5caa7eebfdebb8348e56f000',
  notes: null
};

export const mockScheduleAfterAdd = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  },
  mockPairingToAddToSched
];

export const mockQueryFromgetAvailablePairings = {
         query: `{
    getAvailablePairings(filter: { program: "BE", module: 3, date: "Wed Apr 17 2019" }) {
      pairer {
        name
        module
        program
        id
        pronouns
        skills
        slack
      }
      date
      time
      id
    }
  }`
       };

export const mockQueryFromgetUser = {
         query: `{
  getUser(name: "Hillary") {
    name
    program
    module
    id
  }
}`
       };

export const mockQueryFromgetUserPairings = {
         query: `{
    getUserPairings(
      id: "5h5h5h5h5h5h5"
    ) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromupdatePairing = {
         query: `mutation {
    updatePairing(pairing: {
        id: "6t6t6t"
        pairee: "7b7b7b"
        notes: "Mythical Creatures"
    }) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        id
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromdeletePairing = {
         query: `mutation {
    deletePairing(id: "45dtl") {
      date
    }
  }`
       };