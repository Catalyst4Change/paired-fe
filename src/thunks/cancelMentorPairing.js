import { fetchData } from '../utils';
import { setError, setLoading, cancelPairing } from '../actions';
import * as gql from '../queries';

export const cancelMentorPairing = (pairingId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMentorPairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(cancelPairing(pairingId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
