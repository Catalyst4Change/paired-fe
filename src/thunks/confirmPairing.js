import { fetchData } from '../utils';
import { setError, setLoading, addToSchedule, clearAvailPairings } from '../actions';
import * as gql from '../queries';

export const confirmPairing = (pairingId, paireeId, notes) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const query = gql.updatePairing(pairingId, paireeId, notes);
      const response = await fetchData(query);
      const pairing = await response.updatePairing;
      dispatch(addToSchedule(pairing));
      dispatch(clearAvailPairings());
    } catch(error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};
