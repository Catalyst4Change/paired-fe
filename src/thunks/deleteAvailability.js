import { fetchData } from "../utils";
import { setError, setLoading } from '../actions';
import * as gql from '../queries';

export const deleteAvailability = (userId) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const mutation = gql.deletePairings(userId);
      await fetchData(mutation);
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};