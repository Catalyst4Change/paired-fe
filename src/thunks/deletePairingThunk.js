import { fetchData } from '../utils';
import { setError, setLoading, deletePairing } from '../actions';
import * as gql from '../queries';

export const deletePairingThunk = (pairingId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.deletePairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(deletePairing(pairingId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
