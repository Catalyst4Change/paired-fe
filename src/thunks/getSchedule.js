import { fetchData } from '../utils';
import { setError, setLoading, setSchedule } from '../actions';
import * as gql from '../queries';

export const getSchedule = (id) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const pairingQuery = gql.getUserPairings(id);
      const pairingRes = await fetchData(pairingQuery);
      dispatch(setSchedule(pairingRes.getUserPairings));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
