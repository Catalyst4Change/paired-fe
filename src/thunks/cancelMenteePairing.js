import { fetchData } from '../utils';
import { setError, setLoading } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';

export const cancelMenteePairing = (pairingId, userId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMenteePairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(getSchedule(userId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
