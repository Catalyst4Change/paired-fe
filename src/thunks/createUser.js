import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';
import * as helpers from '../helpers';

export const createUser = (userInfo, availabilities) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.createUser(userInfo);
      const { user } = await fetchData(userQuery);
      const startDate = 'Mon Apr 29 2019';
      const inning = helpers.getDatesToDisplayFancy(startDate, 29);
      const unformattedPairings = helpers.createPairingsForQuery(
        availabilities,
        inning,
        user.id
      );
      const formattedPairings = helpers.formatPairingsForQuery(unformattedPairings)
      const pairingQuery = gql.createPairings(formattedPairings);
      await fetchData(pairingQuery);
      dispatch(setUser(user));
      dispatch(getSchedule(user.id));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};




