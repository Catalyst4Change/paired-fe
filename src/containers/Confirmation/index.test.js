import React from 'react';
import { shallow } from 'enzyme';
import { Confirmation, mapStateToProps, mapDispatchToProps } from './index';
import * as data from '../../mockData';
import { confirmPairing } from '../../thunks/confirmPairing';
import { deletePairingThunk } from '../../thunks/deletePairingThunk'

jest.mock('../../thunks/confirmPairing');
jest.mock('../../thunks/deletePairingThunk');

describe('Confirmation', () => {
  describe('Confirmation component', () => {
    let wrapper;
    const mockDeletePairingThiunk = jest.fn();
    const mockConfirmPairing = jest.fn();
    const mockPropsNoConflict = {
      selectedPairing: '5caa7eebfdebb8348e53a4a4',
      availPairings: data.mockAvailPairings,
      schedule: data.mockSchedule,
      user: data.mockUser,
      confirmPairing: mockConfirmPairing,
      deletePairingThunk: mockDeletePairingThiunk
    };
    beforeEach(() => {
      wrapper = shallow(<Confirmation {...mockPropsNoConflict} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return a props object with selectedPairing, availPairing, schedule and user', () => {
      const expected = {
        selectedPairing: '5caa7eebfdebb8348e53a4a4',
        availPairings: data.mockAvailPairings,
        schedule: data.mockSchedule,
        user: data.mockUser,
      };
      const initialState = {
        selectedPairing: '5caa7eebfdebb8348e53a4a4',
        availPairings: data.mockAvailPairings,
        schedule: data.mockSchedule,
        user: data.mockUser,
        hasError: '',
        isLoading: false
      };
      const mappedProps = mapStateToProps(initialState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mockPairingId = '3904wfjdo';
    const mockPaireeId = '329ruefwjipn';
    const mockNotes = 'Please help with hooks';

    it('should call dispatch with confirmPairing when confirmParing is called', () => {
      const actionToDispatch = confirmPairing(mockPairingId, mockPaireeId, mockNotes);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.confirmPairing(mockPairingId, mockPaireeId, mockNotes);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with deletePairingThunk when deletePairingThunk is called', () => {
      const actionToDispatch = deletePairingThunk(mockPairingId);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deletePairingThunk(mockPairingId);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
