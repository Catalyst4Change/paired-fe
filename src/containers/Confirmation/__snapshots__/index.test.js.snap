// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Confirmation component should match the snapshot 1`] = `
<section
  className="Confirmation"
>
  <ConfirmCard
    confirmPairing={[MockFunction]}
    date="Thu Apr 18 2019"
    deletePairingThunk={[MockFunction]}
    hasOpeningAlready={false}
    name="Tiffany"
    selectedPairing="5caa7eebfdebb8348e53a4a4"
    time="morning"
    userId="5caa7eebfdebb8348e53a48e"
  />
</section>
`;
